release/k3s/images.go

```diff
@@ -0,0 +1,169 @@
+package k3s
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	reg "github.com/rancher/ecm-distro-tools/registry"
+)
+
+type RegistryClient interface {
```

Comment:
In Go, we want to have interfaces named with "er" at the end since interfaces define behavior/verbs. So this would be more appropriately named `Imager`. It's common practice to take the action it performs (since we keep interfaces super small in Go) and add "er" to it.

Reply:
Is there any chance this could be consolidated into a single interface that could be defined here and used for the one below also?